### PR TITLE
fix k8s env var check and reconnect logic

### PR DESCRIPTION
**Pull request by ryant1986**

### CODE REVIEW

**Review by yangxizi030 — APPROVED** [review]
lgtm

File: mmd/compositeconn.go [hunk 1]
```diff
@@ -241,7 +241,7 @@ func (c *CompositeConn) createAndInitDirectConnection(service string) (*ConnImpl
 var env = computeEnv()
 var nameserverUrl = env + ".k8s.peak6.net:53"
 var istioIngressUrl = env + ".istioingress.peak6.net"
-var isInK8s, _ = strconv.ParseBool(os.Getenv("KUBERNETES_SERVICE_HOST"))
+var _, isInK8s = os.LookupEnv("KUBERNETES_SERVICE_HOST")
```
Comment (pkuprys) [inline mmd/compositeconn.go:244]: Is it possible for "KUBERNETES_SERVICE_HOST" to be empty or non-truthy, implying that it is NOT in k8s?

**Review by pkuprys — CHANGES_REQUESTED** [review]
just one question

File: mmd/compositeconn.go (hunk 1 above)
Comment (ryant1986) [reply mmd/compositeconn.go:244]: if that is the case, then `isInK8s` will be `false` which is what I want.  If it's set to anything, `isInK8s` will be `true`, which is what I want.  Am I missing something?

Comment (ryant1986) [reply mmd/compositeconn.go:245]: I see what you're saying.  I don't think it's possible or a case we need to worry about

**Review by pkuprys — APPROVED** [review]
lgtm

**Pull request merged**